cmake/compilers/GNU.cmake

```diff
@@ -72,10 +72,14 @@ endif ()
 # Gnu flags to prevent compiler from optimizing out security checks
 set(TBB_COMMON_COMPILE_FLAGS ${TBB_COMMON_COMPILE_FLAGS} -fno-strict-overflow -fno-delete-null-pointer-checks -fwrapv)
 set(TBB_COMMON_COMPILE_FLAGS ${TBB_COMMON_COMPILE_FLAGS} -Wformat -Wformat-security -Werror=format-security
-	                                                  -fstack-protector-strong -fPIC -shared )
+    -fstack-protector-strong )
 set(TBB_LIB_LINK_FLAGS ${TBB_LIB_LINK_FLAGS} -Wl,-z,relro,-z,now,-z,noexecstack)
 set(TBB_COMMON_COMPILE_FLAGS ${TBB_COMMON_COMPILE_FLAGS} $<$<NOT:$<CONFIG:Debug>>:-D_FORTIFY_SOURCE=2> )
 
+# Text relocation for MIPS doesn't work for heavy tests with too many symbols in debug.
+if (NOT "${CMAKE_SYSTEM_PROCESSOR}" MATCHES "mips")
+    set(TBB_COMMON_COMPILE_FLAGS ${TBB_COMMON_COMPILE_FLAGS} -fPIC -shared)
```

Comment:
Do we really want this `-shared` flag being set explicitly? AFAIK, it is controlled by https://cmake.org/cmake/help/latest/variable/BUILD_SHARED_LIBS.html variable, which is set to `ON` by default in our case. I would suggest just to remove this `-shared` thing from here.

Reply:
Thanks Ilya for pointing this. The reference wiki I was looking at was indicating "-fPIC -shared" as a compile time flag to build the lib as a  position independent code. But you are correct,. I could verify that BUILD_SHARED_LIBS automatically adds -fPIC as a compile flag and -shared as Linker flag - which is sufficient. I have removed those flags all-together.